Use private config repo instead of hard-coded config files

dmolesUC3 opened the pull request:
This pull request updates the Capistrano deploy process to pull from the private [cdlib/mrt-dashboard-config](https://github.com/cdlib/mrt-dashboard-config) repository.

1. if a plain `shared/config` directory exists on the server, it is renamed to `config.old`.
2. if `mrt-dashboard-config` hasn't been cloned:
   - we clone it as `shared/mrt-dashboard-config`
   - we symlink `shared/config` to `shared/mrt-dashboard-config`
3. we use `git fetch` and `git reset --hard` to update the cloned `mrt-dashboard-config`

By default, we use the `master` branch of `mrt-dashboard-config`. We don't prompt for a config tag, but we can specify one, distinct from the tag for the code itself, with the `$CONF_TAG` environment variable, e.g.

```
   CONF_TAG=2018.6.1.alpha TAG=2018.6 cap mrt-ui-dev deploy
```

Note that unlike [Dash](https://github.com/CDLUC3/dashv2/blob/v0.2.30/config/deploy.rb#L78-L87):
- we don't support branches, just tags
- we don't assume the config repo has a tag with the same name as the code repo, or even that it's been tagged at all

mreyescdl commented:
Should we use "git checkout <tag>"  over the reset?
Also, if "config.old" already exists, will another deploy create a conflict of move it in a subdirectory.  Maybe we can use Capistrano ":keep_releases" to hold on to the last few releases.

dmolesUC3 commented:
1. `git checkout` will get confused if there are any local changes, and if we're already on the branch we want (e.g. master) it won't update. `git reset --hard` says "I don't care what state you're in, revision xyz is the source of truth now."
2. A second deploy shouldn't create a conflict because it only does the move if `config` is a directory, and if we've deployed properly `config` will be a symlink.
3. `:keep_releases` only applies to the core application deployment—this is a side effect running some custom shell code outside the main Capistrano deployment and the main Capistrano directories.

   If we think it's important to keep releases, we could put a full Capistrano deploy stack in the config directory (deploying the config as a separate "application"—though we could trigger that from the main application deployment job, too), but I'm not sure it's worth the trouble; if we need to roll back, we can run `deploy:update_config` separately with the previous tag (or tag a previous revision on the fly if we need to).

dmolesUC3 commented:
@mreyescdl—looks like I had you "assigned" instead of "requested review from"; fixed.

@sfisher — I added you as a reviewer in case you have any comments based on the Dash config deployment.
